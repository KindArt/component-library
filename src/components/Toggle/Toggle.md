Colours:

```tsx
<Toggle defaultChecked />
<Toggle defaultChecked colour="BLUE" />
<Toggle defaultChecked colour="ORANGE" />
<Toggle defaultChecked colour="RED" />
```

With Text:

```tsx
<Toggle label="This is a label" description="This is a example description" />
```
