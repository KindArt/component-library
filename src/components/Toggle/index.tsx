import React, { FC } from 'react';
import Checkbox, { ICheckboxProps } from '../Checkbox';

const Toggle: FC<ICheckboxProps> = ({ ...rest }) => <Checkbox toggle {...rest} />;

export default Toggle;
