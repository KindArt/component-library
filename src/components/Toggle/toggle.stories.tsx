import React from 'react';
import Toggle from '.';

export default {
  component: Toggle,
  title: 'Toggle'
};

export const normal = () => <Toggle name="normal" />;
