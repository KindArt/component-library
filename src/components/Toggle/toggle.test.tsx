import React from 'react';
import { shallow } from 'enzyme';
import Toggle from './';

describe('<Toggle />', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallow(<Toggle name="toggle-test" />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Visuals', () => {
    it('should have the toggle prop set to true', () => {
      const actual = wrapper.props().toggle;
      const expected = true;
      expect(actual).toEqual(expected);
    });
  });
});
