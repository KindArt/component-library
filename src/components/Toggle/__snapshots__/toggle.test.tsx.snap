// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should render 1`] = `
<Checkbox
  name="toggle-test"
  toggle={true}
/>
`;
